components/content-service/pkg/storage/gcloud.go

```diff
@@ -889,30 +892,89 @@ func (p *PresignedGCPStorage) DeleteObject(ctx context.Context, bucket string, q
 		err = client.Bucket(bucket).Object(query.Name).Delete(ctx)
 		if err != nil {
 			log.WithField("bucket", bucket).WithField("object", query.Name).Error(err)
+			if err == gcpstorage.ErrBucketNotExist {
+				return ErrNotFound
+			}
 			return err
 		}
 		return nil
 	}
-	if query.Prefix != "" {
-		prefix := query.Prefix
-		if !strings.HasSuffix(prefix, "/") {
-			prefix = prefix + "/"
-		}
 
-		b := client.Bucket(bucket)
-		it := b.Objects(ctx, &storage.Query{
+	prefix := query.Prefix
+	b := client.Bucket(bucket)
+	var it *gcpstorage.ObjectIterator
+	if prefix != "" && prefix != "/" {
+		it = b.Objects(ctx, &storage.Query{
 			Prefix: prefix,
 		})
-		for {
-			attrs, err := it.Next()
-			if err == iterator.Done {
-				break
-			}
-			err = b.Object(attrs.Name).Delete(ctx)
-			if err != nil {
-				log.WithField("bucket", bucket).WithField("object", attrs.Name).Error(err)
-			}
+	} else {
+		it = b.Objects(ctx, nil)
+	}
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		err = b.Object(attrs.Name).Delete(ctx)
+		if err != nil {
+			log.WithField("bucket", bucket).WithField("object", attrs.Name).Error(err)
 		}
 	}
+
+	if err == gcpstorage.ErrBucketNotExist {
+		return ErrNotFound
+	}
 	return err
 }
+
+// DeleteBucket deletes a bucket
+func (p *PresignedGCPStorage) DeleteBucket(ctx context.Context, bucket string) (err error) {
+	client, err := newGCPClient(ctx, p.config)
+	defer client.Close()
+
+	err = p.DeleteObject(ctx, bucket, &DeleteObjectQuery{})
+	if err != nil {
+		if err == gcpstorage.ErrBucketNotExist {
+			return ErrNotFound
+		}
```

Comment:
If we try to delete the object and the object is gone, is that an error?

Reply:
At least this consume would like to know if the object was there:

https://github.com/gitpod-io/gitpod/blob/15d59d9011b50db86ad6a357e1072b3dfcdea67d/components/server/src/workspace/workspace-deletion-service.ts#L82-L90

Reply:
do we do something with this info?

Reply:
Does not seem so. I changed the behavior and do not return NotFound anymore.